File: libs/reservation/src/main/kotlin/io/patterueldev/reservation/currentsong/CurrentSong.kt
package io.patterueldev.reservation.currentsong

abstract class CurrentSong {
    abstract val id: String
    abstract val title: String
    abstract val artist: String
    abstract val thumbnailPath: String
    abstract val videoPath: String
    abstract val durationInSeconds: Int
    abstract val reservingUser: String
    abstract val lyrics: String
    var volume: Double = 1.0
}

File: client/packages/common/lib/song/current_song.dart
part of '../common.dart';

class CurrentSong {
  final String id;
  final String title;
  final String artist;
  final String thumbnailURL;
  final String videoURL;
  final int durationInSeconds;
  final String lyrics;
  final String reservingUser;
  final double volume;

  CurrentSong({
    required this.id,
    required this.title,
    required this.artist,
    required this.thumbnailURL,
    required this.videoURL,
    required this.durationInSeconds,
    required this.lyrics,
    required this.reservingUser,
    required this.volume,
  });
}
